tensorflow-framework/src/main/java/org/tensorflow/framework/activations/Exponential.java

```diff
@@ -49,9 +47,10 @@ public Exponential(Ops tf) {
    *
    * @param input the input tensor
    * @return an Operand for the exponential activation: <code>exp(x)</code>.
+   * @param <T> the data type of the activation
    */
   @Override
-  public Operand<T> call(Operand<T> input) {
+  public <T extends TNumber> Operand<T> call(Operand<T> input) {
```

Comment:
Maybe having the generic parameter at the class level wasn’t a bad idea after all, but it should be only at the base class, i.e. 

`class Exponential extends Activation<TFloating>` and then the method in `class Activation<T extends TNumber>` is declares as  `public <U extends T> Operand<U> ...`,

I’m not in front of my computer so I can’t validate, would that work @JimClarke5 ?

It is always better if we can constrain  the type at compile time rather than at runtime with exceptions.

Reply:
I was able to setup Activation as:
`class Activation<T extends TNumber>`
and
`public abstract <U extends T> Operand<U> call(Operand<U> input);`

Then for ELU:
`ELU extends Activation<TFloating>`
and
`public <U extends TFloating> Operand<U> call(Operand<U> input) {`

Reply:
This is actually a much cleaner implementation and reduces many cast's. 
I still have the issue of checking `TNumber` and `TBool` with initializers `Constant`, `Ones`, and `Zeros`,
but that is not a big deal.

Thanks @karllessard .